src/utilities/routerNavigation.ts

```diff
@@ -0,0 +1,31 @@
+import { isBrowser } from '@/utilities/isBrowser'
+import { updateBrowserUrl } from '@/utilities/updateBrowserUrl'
+
+type RouterNavigation = {
+  forward: () => void,
+  back: () => void,
+  go: (delta: number) => void,
+  update: (url: string) => void,
+}
+
+export function createRouterNavigation(): RouterNavigation {
+  return isBrowser() ? createWebNavigation() : createMemoryNavigation()
+}
+
+function createWebNavigation(): RouterNavigation {
+  return {
+    go: window.history.go,
+    back: window.history.back,
+    forward: window.history.forward,
+    update: updateBrowserUrl,
+  }
+}
+
+function createMemoryNavigation(): RouterNavigation {
+  return {
+    go: () => {},
+    back: () => {},
+    forward: () => {},
+    update: updateBrowserUrl,
+  }
```

Comment:
Can these just throw not implemented for now. IMO we do want the go, forward, and back methods to error. If you're running in node and you call one of these something's wrong and not going to work as expected.